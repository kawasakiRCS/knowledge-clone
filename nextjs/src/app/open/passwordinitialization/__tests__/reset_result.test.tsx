/**
 * パスワードリセット完了ページテスト
 * 
 * @description 旧システムとの互換性テストを含む
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import PasswordResetResultPage from '../reset_result/page';

// useRouter mockの設定
const mockPush = jest.fn();
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

// useLocale mockの設定
jest.mock('@/hooks/useLocale', () => ({
  useLocale: () => ({
    t: (key: string) => {
      const translations: { [key: string]: string } = {
        'knowledge.auth.title.forgot.password': 'Password Reset',
        'knowledge.auth.msg.changed': 'The password has been initialized. Please try to sign in.',
        'knowledge.auth.label.back.top': 'Back to Top',
      };
      return translations[key] || key;
    },
  }),
}));

describe('PasswordResetResultPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('基本レンダリング', () => {
    test('ページタイトルが表示される', () => {
      render(<PasswordResetResultPage />);
      
      const title = screen.getByRole('heading', { level: 4 });
      expect(title).toHaveClass('title');
      expect(title).toHaveTextContent('Password Reset');
    });

    test('完了メッセージが表示される', () => {
      render(<PasswordResetResultPage />);
      
      expect(screen.getByText('The password has been initialized. Please try to sign in.')).toBeInTheDocument();
    });

    test('Topへ戻るリンクが表示される', () => {
      render(<PasswordResetResultPage />);
      
      const backLink = screen.getByRole('link', { name: 'Back to Top' });
      expect(backLink).toHaveAttribute('href', '/');
    });
  });

  describe('レイアウト構造', () => {
    test('適切な余白が設定されている', () => {
      const { container } = render(<PasswordResetResultPage />);
      
      const brElements = container.querySelectorAll('br');
      expect(brElements).toHaveLength(4); // 旧システムと同じ改行数
    });

    test('メッセージ後に適切な余白がある', () => {
      const { container } = render(<PasswordResetResultPage />);
      
      const content = container.firstChild;
      expect(content).toMatchSnapshot();
    });
  });

  describe('旧システム互換性', () => {
    test('見た目が旧システムと同等である', () => {
      const { container } = render(<PasswordResetResultPage />);
      
      // タイトルのクラス
      const title = container.querySelector('h4.title');
      expect(title).toBeInTheDocument();
      
      // Topへ戻るリンク
      const link = container.querySelector('a[href="/"]');
      expect(link).toBeInTheDocument();
    });
  });
});