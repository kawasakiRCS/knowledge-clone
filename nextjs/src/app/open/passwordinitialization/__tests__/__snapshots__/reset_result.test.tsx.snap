// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PasswordResetResultPage レイアウト構造 メッセージ後に適切な余白がある 1`] = `
<h4
  class="title"
>
  Password Reset
</h4>
`;
